src/coreComponents/mesh/mpiCommunications/NeighborCommunicator.cpp

```diff
@@ -620,5 +620,159 @@ void NeighborCommunicator::unpackBufferForSync( std::map< string, string_array >
   }
 }
 
+///////////////////////////  New versions of the packing and unpacking functions //////////////////////////////////////////////////////////////
+int NeighborCommunicator::packCommSizeForSync( std::map< fieldSyncKey, string_array > const & fieldsTobeSync,
+                                               MeshLevel const & mesh,
+                                               int const commID,
+                                               bool onDevice,
+                                               parallelDeviceEvents & events )
+{
+  GEOSX_MARK_FUNCTION;
+
+  NodeManager const & nodeManager = mesh.getNodeManager();
+  EdgeManager const & edgeManager = mesh.getEdgeManager();
+  FaceManager const & faceManager = mesh.getFaceManager();
+  ElementRegionManager const & elemManager = mesh.getElemManager();
+
+  arrayView1d< localIndex const > const & nodeGhostsToSend = nodeManager.getNeighborData( m_neighborRank ).ghostsToSend();
+  arrayView1d< localIndex const > const & edgeGhostsToSend = edgeManager.getNeighborData( m_neighborRank ).ghostsToSend();
+  arrayView1d< localIndex const > const & faceGhostsToSend = faceManager.getNeighborData( m_neighborRank ).ghostsToSend();
+
+  int bufferSize = 0;
+
+  for( auto const & fields : fieldsTobeSync )
+  {
+    fieldSyncKey key = fields.first;
+    switch( key.location )
+    {
+      case Location::Node:
+      {
+        bufferSize += nodeManager.packSize( fields.second, nodeGhostsToSend, 0, onDevice, events );
+      }
```

Comment:
```suggestion
      {
        bufferSize += nodeManager.packSize( fields.second, nodeGhostsToSend, 0, onDevice, events );
        break;
      }
```
Important for all your switches.